examples/flax/language-modeling/run_bart_dlm_flax.py

```diff
@@ -888,7 +894,7 @@ def eval_step(params, batch):
                 num_eval_samples = len(tokenized_datasets["validation"])
                 # Avoid using jax.numpy here in case of TPU training
                 eval_samples_idx = np.arange(num_eval_samples)
-                eval_batch_idx = generate_batch_splits(eval_samples_idx, eval_batch_size, drop_last=False)
+                eval_batch_idx = generate_batch_splits(eval_samples_idx, eval_batch_size)
```

Comment:
Unintentional change?

Reply:
@sanchit-gandhi I noticed that with `drop_last=False` the eval loss will become `nan` at the beginning of training, but eval accuracy is still on track. It appears to occur with both `run_bart_dlm_flax` and `run_summarization` so I temporarily turned it off. It would be great if you could take a look and fix it.

Reply:
Interesting! What immediately jumps out to me is that `num_labels` is 0, causing the 'true loss' to be `nan`. You didn't get this behaviour previously with the `pmap` operation? What eval batch size are you using?

Reply:
This bug occurs without dividing `loss` by `num_labels` actually. I took a quick look at the eval loss computed on every token at each step, the last losses was `nan`. 

```bash
python run_summarization_flax.py \
	--output_dir ./bart-base-xsum \
	--model_name_or_path facebook/bart-base \
	--tokenizer_name facebook/bart-base \
	--dataset_name="xsum" \
	--do_train --do_eval --do_predict --predict_with_generate \
	--num_train_epochs 6 \
	--learning_rate 5e-5 --warmup_steps 0 \
	--per_device_train_batch_size 64 \
	--per_device_eval_batch_size 64 \
	--overwrite_output_dir \
	--max_source_length 512 --max_target_length 64 \
	--push_to_hub 
```
Printed tensor has a shape of (8, 64, 64), provided that I have 8 TPU cores, `per_device_eval_batch_size=64` and `max_target_length=64`.
![ảnh](https://user-images.githubusercontent.com/38061659/184084271-4100a7a0-966f-4f0e-bddb-dfc604275254.png)

Reply:
Oh no! That's interesting to see. Is this an artefact of using the `psum`? As in, were the losses `nan` when we used a `pmap` previously? If so, we'll need to address this!

Reply:
The losses were `nan` right after `loss = optax.softmax_cross_entropy(logits, soft_labels)`. Great if you could have a look at this issue!

Reply:
Anyway, this problem doesn't seem to be related to this PR. Flax examples are young so we'll improve it step by step :)

Reply:
Hi @duongna21, just wondering what the status is on the `nan` issue being discussed here. I'm running into this issue while using the recently added `run_bart_dlm_flax.py` script but am very new to Flax/Jax so haven't been able to really make sense of it yet. Is `run_bart_dlm_flax.py` useable for model pre-training in its current state? Just as a side note, I haven't seen the `nan` issue when running `run_t5_mlm_flax.py` on my training data. Thanks in advance for any clarification!